cspreports/models.py

```diff
@@ -77,12 +77,22 @@ class Meta(object):
     violated_directive = models.TextField(blank=True, null=True)
     original_policy = models.TextField(blank=True, null=True)
     effective_directive = models.TextField(blank=True, null=True)
-    status_code = models.PositiveSmallIntegerField(blank=True, null=True)
+    status_code = models.PositiveSmallIntegerField(blank=True, null=True, validators=[MinValueValidator(0)])
     source_file = models.TextField(blank=True, null=True)
-    line_number = models.PositiveIntegerField(blank=True, null=True)
-    column_number = models.PositiveIntegerField(blank=True, null=True)
+    line_number = models.PositiveIntegerField(blank=True, null=True, validators=[MinValueValidator(0)])
+    column_number = models.PositiveIntegerField(blank=True, null=True, validators=[MinValueValidator(0)])
     disposition = models.CharField(max_length=10, blank=True, null=True, choices=DISPOSITIONS)
 
+    def clean(self):
+        super(CSPReport, self).clean()
+        for field in REQUIRED_FIELDS:
+            django_field_name = field.replace("-", "_")
+            model_field = getattr(self, django_field_name)
+            if model_field is None or model_field == '':
+                self.is_valid = False
+                return
+        self.is_valid = True
```

Comment:
I wouldn't modify the instance by a `clean` method. That just looks like a unwanted side effect.